src/components/Transaction/CheckoutForm.tsx

```diff
@@ -0,0 +1,134 @@
+import { Button } from '@mantine/core'
+import { PaymentElement, useStripe, useElements } from '@stripe/react-stripe-js'
+import React, { Dispatch, SetStateAction, useState } from 'react'
+import Swal from 'sweetalert2'
+import Stripe from 'stripe'
+import { proceedToPayment } from '../../service/TransactionService'
+
+interface CheckoutFormProps {
+  intentId: string
+  total: number
+  conversionRate: string
+  setResetPayment: Dispatch<SetStateAction<boolean>>
+  setShouldUpdate: Dispatch<SetStateAction<boolean>>
+  setValue: Dispatch<SetStateAction<string>>
+}
+export default function CheckoutForm({
+  intentId,
+  total,
+  conversionRate,
+  setResetPayment,
+  setShouldUpdate,
+  setValue,
+}: CheckoutFormProps) {
+  const stripe = useStripe()
+  const elements = useElements()
+  const stripeObj = new Stripe(process.env.REACT_APP_STRIPE_SECRET_KEY!, {
+    apiVersion: '2023-10-16',
+  })
+
+  const [message, setMessage] = useState('')
+  const [isProcessing, setIsProcessing] = useState(false)
+
+  const handleProceedToPayment = async () => {
+    const data = await proceedToPayment(
+      total / Number(conversionRate),
+      `You purchased ${total / Number(conversionRate)} credits for $${total}.`,
+    )
+    console.log(data)
+  }
+
+  const handleSubmit = async (e: any) => {
+    e.preventDefault()
+
+    if (!stripe || !elements) {
+      return
+    }
+
+    setIsProcessing(true)
+
+    const response = await stripe.confirmPayment({
+      elements,
+      confirmParams: {
+        // return_url: `${window.location}`,
+      },
+      redirect: 'if_required',
+    })
+    if (response.error) {
+      setMessage(response.error.message as string)
+      setIsProcessing(false)
+    } else {
+      await handleProceedToPayment()
+      Swal.fire({
+        position: 'center',
+        icon: 'success',
+        title: 'Payment successful!',
+        showConfirmButton: false,
+        timer: 1500,
+      })
+      setResetPayment((prev) => !prev)
+      setShouldUpdate((prev) => !prev)
+      setValue('Exchange Policy')
+    }
+
+    // const { error } = await stripe.confirmPayment({
```

Comment:
remove unused code or Comment the reason for keeping it

Reply:
fixed